src/protocol.rs

```diff
@@ -190,6 +190,62 @@ pub fn read_info_field(device: &HidDevice, field: InfoField) -> Result<String, E
         .to_owned())
 }
 
+pub fn run_tap_commands(device: &HidDevice) -> Result<(), Error> {
+    const TAP_REPORT_ID: u8 = 2;
+    const TAP_REPORT_LEN: usize = 2048;
+
+    loop {
+        print!("> ");
+        std::io::stdout().flush().unwrap();
+        let mut tap_command = String::new();
+        std::io::stdin()
+            .read_line(&mut tap_command)
```

Comment:
Might be worth pulling in something like [rustyline](https://github.com/kkawakam/rustyline) here, if it doesn't bloat the binary size too much. Interactive shells without history or line editing are very frustrating, in my experiece.

Then again, the commands are four characters at most, so maybe it's not worth the effort. I'm happy to merge as-is and then bring in rustyline in a follow-up commit myself if it bothers me.